Formula/r/recc.rb

```diff
@@ -0,0 +1,231 @@
+  depends_on "tomlplusplus" => :build
+  depends_on "abseil"
+  depends_on "c-ares"
+  depends_on "glog"
+  depends_on "grpc"
+  depends_on "openssl@3"
+  depends_on "protobuf"
+  depends_on "re2"
+
+  on_macos do
+    depends_on "gflags"
+  end
+
+  on_linux do
+    depends_on "pkg-config" => :build
+    depends_on "util-linux" => :build
+    depends_on "zlib"
+  end
+
+  def platform_os
+    on_macos do
+      "macos"
+    end
+    on_linux do
+      "linux"
+    end
+  end
+
+  def generate_recc_conf(config_directory)
+    recc_conf_tmpl = <<~EOS
+      ##
+      # For the list of configuration parameters please visit:
+      #   https://buildgrid.gitlab.io/recc/configuration-variables.html
+      #
+      # The configuration file settings use lowercase letters without the RECC_ prefix
+      # By default recc reads the configuration options from the following places,
+      # applying settings bottom-up, with 1 being the last applied configuration.
+      # If an option is set in multiple places, the one higher on this list will be the
+      # effective one
+      #
+      # 1. Environment variables
+      # 2. ${cwd}/recc/recc.conf
+      # 3. {$RECC_CONFIG_DIRECTORY}/recc.conf
+      # 4. ~/.recc/recc.conf
+      # 5. <%= "#{etc}/recc/recc.conf" %>
+      # 6. <%= "#{prefix}/etc/recc/recc.conf" %>
+      ##
+
+      # Protocol version used for communicating with the REAPI server
+      reapi_version=2.2
+
+      # The URI of the server
+      server=<%= "unix://#{var}/recc/casd/casd.sock" %>
+      # Instance name provided by the server
+      instance=recc-server
+
+      # Identifier to correlate all recc invocations#{" "}
+      correlated_invocations_id=<%= #{`hostname`.strip} %>
+
+      # Use cache-only mode, build anything not available in the cache locally
+      cache_only=1
+      # Upload results from the local builds to cache
+      cache_upload_local_build=1
+      # Upload failed results from the local builds to cache
+      cache_upload_failed_build=1
+
+      # Enable for link commands#{" "}
+      link=1
+      # Use cache-only mode for link commands#{" "}
+      link_cache_only=1
+
+      # Report all entries returned by the dependency command, even if they are absolute paths
+      deps_global_paths=1
+      # Do not rewrite absolute paths to be relative.
+      no_path_rewrite=1
+
+      # Do not retry#{" "}
+      retry_limit=0
+
+      # Platform properties
+      REMOTE_PLATFORM_OS=<%= platform_os %>
+      REMOTE_PLATFORM_ISA=<%= Hardware::CPU.arch.to_s %>
+    EOS
+
+    config_directory.mkpath
+    File.write(config_directory / "recc.conf", ERB.new(recc_conf_tmpl).result(binding))
+  end
+
+  def generate_recc_server(bin_directory)
+    (bin_directory / "recc-server").write <<~EOS
+      #!/usr/bin/env sh
+      #{bin}/buildbox-casd \
+        --local-server-instance=recc-server \
+        --protect-session-blobs \
+        #{var}/recc/casd
+    EOS
+  end
+
+  def generate_recc_wrapper(bin_directory, compiler)
```

Comment:
```suggestion
  def generate_recc_wrapper(compiler)
```
This should also be in-lined in the `install` method.